content-resources/src/main/java/flyway/oskari/V3_1_1__remove_analysis_from_appsetups.java

```diff
@@ -0,0 +1,205 @@
+        LOG.debug(OffsetDateTime.now().format(dtf) + " Starting update.");
+        for (int i = 0, counter = 0; i < mapfullConfigs.length(); i++, counter++) {
+            JSONObject mapfull = mapfullConfigs.optJSONObject(i);
+            JSONObject state = new JSONObject(mapfull.getString("state"));
+
+            if (state == null || !state.has("selectedLayers")) {
+                continue;
+            }
+
+            JSONArray selectedLayers = Optional.ofNullable(state.getJSONArray("selectedLayers")).orElse(new JSONArray());
+            JSONArray newSelectedLayers = new JSONArray();
+            for (int j = 0; j < selectedLayers.length(); j++) {
+                JSONObject layer = selectedLayers.optJSONObject(j);
+                if (layer != null) {
+                    String id = layer.get("id").toString();
+                    if (!id.startsWith(ANALYSIS_LAYER_PREFIX)) {
+                        newSelectedLayers.put(layer);
+                    }
+                }
+            }
+            state.put("selectedLayers", newSelectedLayers);
+
+            int appsetupId = mapfull.getInt("appsetup_id");
+            int bundleId = mapfull.getInt("bundle_id");
+            String valueString = "(" + appsetupId + ", " + bundleId + ", '" + state.toString() + "')";
+            values.add(valueString);
+
+            // execute once a batch size or when we reach the end
+            if (counter == BATCH_SIZE - 1 || i == mapfullConfigs.length() - 1) {
+                String query = sql.replace("${values}", String.join(",\n", values));
+                statement.addBatch(query);
+                counter = 0;
+                values.clear();;
+            }
+        }
+
+        int[] results = statement.executeBatch();
+        LOG.debug("results: ", results.length);
+        LOG.debug(OffsetDateTime.now().format(dtf) + " Finished removing analysis layers from " + mapfullConfigs.length() + " views.");
+    }
+
+
+    private void removeAnalysisLayerPluginFromMapmodules(Connection conn) throws SQLException {
+        JSONArray mapfullConfigs = getMapfullConfigsWithAnalysis(conn);
+        String sql = """
+            UPDATE oskari_appsetup_bundles bundles SET config = v.config 
+            FROM (VALUES
+                ${values}
+            ) AS v(appsetup_id, bundle_id, config)
+            WHERE bundles.appsetup_id = v.appsetup_id AND bundles.bundle_id = v.bundle_id
+        """;
+
+        List<String> values = new ArrayList<>();
+
+        Statement statement = conn.createStatement();
+
+        DateTimeFormatter dtf = DateTimeFormatter.ofPattern("dd.MM.YYYY HH:mm:ss");
+        LOG.debug(OffsetDateTime.now().format(dtf) + " Starting update.");
+        for (int i = 0, counter = 0; i < mapfullConfigs.length(); i++, counter++) {
+            JSONObject mapfull = mapfullConfigs.optJSONObject(i);
+            JSONObject config = new JSONObject(mapfull.getString("config"));
+            if (config != null) {
```

Comment:
JSONObject configcan't be null, but new JSONObject(mapfull.getString("config")); could probably throw an exception if mapfull.getString("config") is null